activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/JobExecutorIT.java

```diff
@@ -466,6 +518,75 @@ public void testBoundaryTimerEvent() throws Exception {
         verify(jobMessageHandler).handleMessage(ArgumentMatchers.<Message<?>>any());
     }
     
+    @Test
+    public void shouldPublishJobMessageFailedEvent() throws InterruptedException {
+        // given
+        CountDownLatch eventPublished = new CountDownLatch(1);
+        String destination = "spyAsyncExecutorJobs";
+        
+        applicationContext.addApplicationListener(new CountDownLatchApplicationEventListener<JobMessageFailedEvent>(eventPublished));
+        
+        doReturn(false).when(spyJobMessageChannel)
+                       .send(ArgumentMatchers.<Message<?>>any());
+        
+        // when
+        new TransactionTemplate(transactionManager).execute(new TransactionCallbackWithoutResult() {
+
+            @Override
+            protected void doInTransactionWithoutResult(TransactionStatus status) {
+                Job job = new JobEntityImpl() {{
+                    setId("jobId");
+                }};
+                
+                jobMessageProducer.sendMessage(destination, job);
+            }
+
+        });
+        
+        // then
+        assertThat(eventPublished.await(1, TimeUnit.SECONDS)).as("should publish JobMessageFailedEvent")
+                                                             .isTrue();
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Please, prefer `catchThrowable` from AssertJ to check the exception. There is an example here https://github.com/Activiti/activiti-core-common/blob/develop/activiti-spring-connector/src/test/java/org/activiti/core/common/spring/connector/ConnectorDefinitionServiceTest.java#L58-L65

Reply:
fixed